modules/Auth/include/ReservationHandler.hpp

```diff
@@ -1,94 +1,355 @@
+    ///
+    /// \brief Constructor.
+    ///
+    ReservationHandler(const std::string& id, kvsIntf* store);
+
+    ///
+    /// \brief Destructor.
+    ///
+    ~ReservationHandler();
+
+    ///
+    /// \brief Add a connector.
+    /// \param evse_id          The evse id of the connector.
+    /// \param connector_id     The connector id.
+    /// \param connector_type   The connector type.
+    /// \param connector_state  The current connector state (default `AVAILABLE`).
+    ///
+    void add_connector(const uint32_t evse_id, const uint32_t connector_id,
+                       const types::evse_manager::ConnectorTypeEnum connector_type,
+                       const ConnectorState connector_state = ConnectorState::AVAILABLE);
+
+    ///
+    /// \brief Initialize. Read reservations from persistent store.
+    ///
+    void init();
+
+    ///
+    /// \brief Try to make a reservation.
+    /// \param evse_id      Optional, the evse id. If omitted, a 'global' reservation will be made.
+    /// \param reservation  The reservation to make.
+    /// \return The result of the reservation (`Accepted` if the reservation could be made).
+    ///
+    types::reservation::ReservationResult make_reservation(const std::optional<uint32_t> evse_id,
+                                                           const types::reservation::Reservation& reservation);
+
+    ///
+    /// \brief Change the EVSE state.
+    ///
+    /// This is important for the reservation handler, to know which EVSE is in which state, to know if a reservation
+    /// can be made or not.
+    ///
+    /// \param state        The EVSE state.
+    /// \param evse_id      The EVSE id.
+    ///
+    void set_evse_state(ConnectorState state, const uint32_t evse_id);
+
+    ///
+    /// \brief Change a specific connector state.
+    ///
+    /// This is important for the reservation handler, to know which connector is in which state, to know if a
+    /// reservation can be made or not.
+    ///
+    /// \param connector_state  The state of the connector.
+    /// \param evse_id          The EVSE id the connector belongs to.
+    /// \param connector_id     The connector id.
+    ///
+    void set_connector_state(ConnectorState connector_state, const uint32_t evse_id, const uint32_t connector_id);
+
+    ///
+    /// \brief Check if charging is possible on a given EVSE.
+    ///
+    /// If there are multiple global reservations, while a charging station might look available, it is possible that
+    /// charging is not possible because then cars that made reservations can not charge anymore.
+    ///
+    /// Only use this function to check if a car can charge without having a reservation id.
+    ///
+    /// \param evse_id  The evse on which a car wants to charge.
+    /// \return True if charging is possible.
+    ///
+    bool is_charging_possible(const uint32_t evse_id);
```

Comment:
It looks like this function is never used except in tests

Reply:
Good one and interesting, because I think this needs implementation then. So if there are two reservations and three charging stations and a car comes in and plugs in, it is possible that it can not charge on a specific evse because then the reservations can not be fulfilled anymore. But I have to think about where to implement this then. Thanks for pointing this out!

Reply:
I added two calls to this function in AuthHandler and created two tests for that (auth_tests) as well. Can you take a look if that's how it should work?